cnd/src/network.rs

```diff
@@ -301,34 +304,27 @@ impl ComitNode {
 
     pub fn initiate_communication(
         &mut self,
-        id: LocalSwapId,
-        swap_params: Herc20HalightBitcoinCreateSwapParams,
+        local_swap_id: LocalSwapId,
+        peer: DialInformation,
+        role: Role,
+        digest: SwapDigest,
+        identities: Identities,
     ) -> anyhow::Result<()> {
-        self.supports_halight()?;
-        self.comit.initiate_communication(id, swap_params)
-    }
-
-    fn init_hbit_herc20(
-        &mut self,
-        id: LocalSwapId,
-        swap: CreatedSwap<hbit::CreatedSwap, herc20::CreatedSwap>,
-    ) -> anyhow::Result<()> {
-        self.comit.init_hbit_herc20(id, swap)
-    }
-
-    fn init_herc20_hbit(
-        &mut self,
-        id: LocalSwapId,
-        swap: CreatedSwap<herc20::CreatedSwap, hbit::CreatedSwap>,
-    ) -> anyhow::Result<()> {
-        self.comit.init_herc20_hbit(id, swap)
-    }
+        let data = match role {
```

Comment:
We match inside `initiate_communication` as-well right?
Could also just make two functions then it is not redundant :)

Reply:
Nice idea, implemented as suggested.